docs/assignment2.md

```diff
@@ -1 +1,322 @@
+    * calls the snippets edit method.
+* _filter(wordToContain:String, tags:String[*], language:String): HashMap<Integer, Snippet>_
+    * filters the snippets managed and returns an integer-snippet map.
+* _getNextId(): Int_
+    * gets the next available id for a new snippet.
+* _isValidId(id:Integer): Boolean_
+    * checks if an integer is a valid id for an existing snippet.
+* _generatePathToSnippetJson(snippetId:Integer): String_
+    * returns the relative path for the requested snippet (by id).
+* _getSnippets(): HashMap<Integer; Snippet>_
+    * returns all the managed snippets as an integer-snippet hashmap.
+#### Associations
+* _SnippetManager < Main_
+    *  On the start of the program the main class constructs a snippetmanager class which maintains all the operations on the snippets.
+* _SnippetManager > Snippet_
+    * described
+* _SnippetManager > JsonIO_
+    * described
+    
+### class: JsonIO
+This class takes care of the conversion between string type and Json type using the GSON library.
+#### Attributes
+* _g: Gson_
+    * contains the Gson object
+#### Operations
+* _JsonIO(): void_
+    * instantiatest a Gson object.
+* _getInstance(): JsonIO_
+    * returns the singleton json object.
+* _writeToJson(pathToJson:String, object :Object): void_
+    * serializes an object.
+* _loadFromJson(pathToJson:String): Snippet_
+    * deserializes a Snippet.
+* _onException(e:Exception): void_
+    * handles exceptions like file not found or bad data.
+#### Associations
+* _JsonIO < Main_
+    * already described
+* _JsonIO < Snippet_
+    * The snippet class uses the JsonIO class to convert snippet objects to json, to store snippet objects.
+* _JsonIO < SnippetManager_
+    * The snippet manager class uses the JsonIO class to convert json to objects, to retrieve snippet objects.
+
+### class: Main
+ 
+#### Attributes
+* _snippetManager: SnippetManager_
+    * the snippet manager used by the current instance of the app.
+* _cliUI: CliUI_
+    * the cliUI used by the current instance of the app.
+#### Operations
+* _main(args:String[*]): void_
+    * Instantiates SnippetManager, CliUI and JsonIO. After this, CliUI handles the flow of the app.
+#### Associations
+* _Main > JsonIO_
+    * instantiates.
+* _Main > SnippetManager_
+    * instantiates.
+* _Main > CliUI_
+    * instantiates.
+    
+## Object diagrams								
+Author(s): Yael Goede
+
+![snapshot Diagram](snapshotDiagram.png)
+In the diagram above a snapshot of the system is shown where the database class is filled with 4 snippets. All snippets have a unique id which makes them unique from each other. Further more all fields are filled with the required information, such as the language, tags and content field. 
+The content field contains the actual code used as snippet by the user. The language field specifies the programming language which is valid for the content field. This field enables features such as the text highlighting feature, which is language specific.
+
```

Comment:
-Objects in an Object diagram should be instances of classes present in the Class diagram
-Snippet objects are ok but there is no SnippetDB class
-Don't model compositions in an object diagram
-Use ":" and Class name to indicate the the class instance i.e Snippet1:Snippet